pkg/costmodel/client.go

```diff
@@ -206,6 +236,17 @@ func (c *Client) parseResults(results model.Value) (Cost, error) {
 			// This is when there is no spot/non-spot label
 			cost.Dollars = value
 		}
+		// Handles the case for cloudcost exporter metrics where `price_tier` is the label for spot/non-spot
+		// TODO: Delete after removing support for OpenCost
+		switch sample.Metric["price_tier"] {
+		case "ondemand":
+			cost.NonSpot = value
+		case "spot":
+			cost.Spot = value
+		default:
+			// This is when there is no spot/non-spot label
+			cost.Dollars = value
+		}
```

Comment:
This is really the heart of the change. The biggest difference between the queries is that the `price_tier` label is used to signify if the price is for an ondemand or spot instance.